### ISSUE TITLE
Metro not detecting that i launched the app

### ISSUE THREAD

**Issue by Snouzy** [facebook/react-native#41324]
### New Version

0.71.4

### Old Version

0.71.0

### Build Target(s)

iOS simulator

### Output of `react-native info`

```
System:
    OS: macOS 13.4
    CPU: (10) arm64 Apple M1 Pro
    Memory: 1.74 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.16.0 - ~/.nvm/versions/node/v16.16.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v12.22.12/bin/yarn
    npm: 8.11.0 - ~/.nvm/versions/node/v16.16.0/bin/npm
    Watchman: 2023.10.09.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.14.2 - /Users/mathiasbradiceanu/.rbenv/shims/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    Android SDK:
      API Levels: 30, 31, 32, 33
      Build Tools: 28.0.3, 29.0.2, 30.0.2, 30.0.3, 33.0.0
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.3 AI-213.7172.25.2113.9123335
    Xcode: 14.3/14E222b - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.11 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0 
    react-native: 0.71.4 => 0.71.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Issue and Reproduction Steps

I am reaching out to report a challenge I’ve encountered after upgrading our application to React Native version 0.71.4. Since the update from version 0.63.3, which was part of a substantial migration carried out last week, I've noticed a significant change in the behavior of the Metro bundler when launching the application using npx react-native run-ios.

The issue manifests as the absence of the usual progress bar in the Metro bundler, which typically indicates the bundling process. Consequently, I am unable to view any console logs, and features such as hot reloading are no longer functioning. This hinders the development process as I cannot receive immediate feedback from my application.

I've gone through the migration process diligently, but I am concerned that there may have been an oversight or a step that I missed. I am seeking assistance to understand the potential causes of this problem and would appreciate any guidance on how to resolve it.

Thank you in advance for your support. 😇

**Comment by alfonsocj** [comment]
👋 @Snouzy, is there a way you could create a reproducible example that others can use? Unfortunately is very difficult to debug an issue without being able to test it locally.

**Comment by Snouzy** [comment]
@alfonsocj Hi, after a day of research, I just found that in 

`npx react-native run-ios --scheme=myApp-staging --simulator='iPhone 14 Pro'`

the scheme was set to `-configuration Staging`, which used to work and linked the app to the Metro bundler fine before my react-native update. 
After I updated to react-native 0.74.4, this was no longer the case.

However, I can't say whether before, when I ran the same command, the configuration was launched in Debug mode or not. 
I haven't updated XCode.

Thanks for your reply.
